7 build nsis khiops nocode installer in the ci

folmos-at-orange opened the pull request:
This PR implements the NSIS installer creation with Github workflow. The workflow is set to be executed manually at the moment and can generate an installer from any branch or tag.

I also added two minor fixes:
- The khiops JAR generated with CMake did not include properly images and icons
  - This made make minor changes within `src` (@marcboulle could check those)
- Minor improvements for the unit test workflow.

Also @marcboulle if you could test the installer generated (https://github.com/KhiopsML/khiops/suites/15487352576/artifacts/884460194)

folmos-at-orange committed:
Fix CMake not building the JARs properly

The khiops.jar contains only icons/images. They were in the "empty"
namespace but the CMake add_jar directive does not work well when the
namespace is empty.  The solution is to always use a namespace for the jar images.

Additionally the add_jar code was refactored to add_khiops_jar function
in the top CMakeLists.txt

folmos-at-orange committed:
Improve unit test workflow

- Add path filtering so it only executs when relevant code is changed
- Use temporarily phoenix-actions/test-reporting until dorny/test-report
  fix a bug showing the dashboard in other workflows
- Improve readability of test steps

marcboulle reviewed (commented):
Remarques sur le test de l'installeur:
- j'ai downloadé un fichier zip (khiops-installer.zip), et non un exe d'installeur
- une fois dézippé, j'ai obtenu un exe au nom tres "technique": khiops-10.1.5-SNAPSHOT-c184f8b-setup.exe, avec ensuite des messages d'installation bases sur ce nom (pas vraiment user friendly)
- lors de l'installation, j'ai eu "editeur inconnu": l'exe ne semble pas signe
- le message concernant l'installation de MPI donne une version tres detaillee de la version

Remarques sur l'installation:
- j'ai un fichier LICENCE (et non LICENCE.txt): on ne peut inspecter son contenu par un simple clic sur le fichier
- disparition des fichiers et repertoires suivants:
  - README.txt
  - install.txt
  - whatsnewV10.1.txt
  - doc
  - samples

Remarques sur l'executable:
- le panneau d'aide (Help/Documentation), pour Khiops et Khiops coclustering référence toujours la doc et les samples

Remarque sur la procédure d'installation
- le README.md ne semble pas suffisant pour reproduire l'installation: l'objectif est que n'importe qui soit autonome pour pouvoir maintenir l'installeur, en maintenance corrective ou évolutives
  - prérequis: l'outil NSIS doit etre installe
  - ou se trouve les installeurs prerequis (java, MPI, visu): "somewhere in your system"???
    - poser les installeurs java et MPI sur un file system partage?
    - référencer les repos des outils de visualisation?
  - comment gerer la signature de l'installeur (a automatiser a terme)
  - comment gerer efficacement le cycle de developpement/test de l'installeur sur sa machine locale, avant de passer via un docker pour le CI/CD?

bruno-at-orange reviewed (changes requested):
few things to update, and things to discuss

bruno-at-orange reviewed (changes requested):
Just few things

folmos-at-orange committed:
Make test khiops install action cross platform

folmos-at-orange committed:
Put mpi suffix in MODL target_link_libraries

folmos-at-orange committed:
Use pull_request trigger in workflows

A pull_request event is better suited than push event for our purposes.
A push event is triggered not only when pushing but when merging via the
GH web interface.

folmos-at-orange committed:
Update position independent code settings

The position independent code settings are since CMake 3.14 handled with
the POSITION_INDEPENDENT_CODE target property. This automatically sets
the proper linker flags to achieve this objective for all supported
toolchains.

folmos-at-orange committed:
Change contact email to khiops.team

folmos-at-orange committed:
Make presets defaults coherent with CMakeLists.txt

folmos-at-orange committed:
Add NSIS windows packaging

- Add and readapt the NSIS scripts
  - All bundled executables are now passed as parameters within the
    makensis call
- Add a workflow that build the NSIS packages in the GH CI
- Update samples location in documentation
- Update readme and whatsnew files

folmos-at-orange committed:
Add version helper scripts

bruno-at-orange reviewed (dismissed):
pourquoi modifier tout le fichier pour ajouter seulement ${MPI_SUFFIX}
Le fichier était très bien comme ça.
Maintenant on ne voit pas ce que tu as modifié, la reviue est impossible...

folmos-at-orange committed:
Update copyright and website labels

marcboulle reviewed (changes requested):
Concernant le suffixe preview, on peut le laisser pour l'instant dans la version windows, mais créer une issue qui:
- pose la question de l'intérêt de ce suffixe
- si ok: homégénisation sur toutes les platformes (linux...)
- si ko: on revient à avant

L'alternative (supression ou commentaire maintenant, puis issue) est plus coûteuse à mettre en place, et ne changera rien au statut après discussion de l'issue.
